LoopKitUI/View Controllers/InsulinSensitivityScheduleViewController.swift

```diff
@@ -0,0 +1,397 @@
+//
+//  InsulinSensitivityScheduleViewController.swift
+//  LoopKitUI
+//
+//  Created by Pete Schwamb on 7/2/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import UIKit
+import HealthKit
+import LoopKit
+
+public enum SaveInsulinSensitivityScheduleResult<T: RawRepresentable> {
+    case success
+    case failure(Error)
+}
+
+public protocol InsulinSensitivityScheduleStorageDelegate {
+    func saveSchedule(_ schedule: InsulinSensitivitySchedule, for viewController: InsulinSensitivityScheduleViewController, completion: @escaping (_ result: SaveInsulinSensitivityScheduleResult<Double>) -> Void)
+}
+
+public class InsulinSensitivityScheduleViewController : DailyValueScheduleTableViewController {
+
+    public init(allowedValues: [Double], minimumTimeInterval: TimeInterval? = nil) {
+        self.allowedValues = allowedValues
+        self.minimumTimeInterval = minimumTimeInterval ?? .minutes(30)
+
+        super.init(style: .grouped)
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+
+        tableView.register(SetConstrainedScheduleEntryTableViewCell.nib(), forCellReuseIdentifier: SetConstrainedScheduleEntryTableViewCell.className)
+        tableView.register(TextButtonTableViewCell.self, forCellReuseIdentifier: TextButtonTableViewCell.className)
+        updateEditButton()
+    }
+
+    @objc private func cancel(_ sender: Any?) {
+        self.navigationController?.popViewController(animated: true)
+    }
+
+    // MARK: - State
+
+    public var insulinSensitivityScheduleStorageDelegate: InsulinSensitivityScheduleStorageDelegate?
+
+    public var unit: HKUnit = HKUnit.milligramsPerDeciliter.unitDivided(by: .internationalUnit())
+
+    public var schedule: InsulinSensitivitySchedule? {
+        get {
+            let validEntries = internalItems.compactMap { (item) -> RepeatingScheduleValue<Double>? in
+                guard let value = item.value else {
+                    return nil
+                }
+                return RepeatingScheduleValue(startTime: item.startTime, value: value)
+            }
+            return InsulinSensitivitySchedule(unit: unit, dailyItems: validEntries, timeZone: timeZone)
+        }
+        set {
+            if let newValue = newValue {
```

Comment:
If `newValue` is `nil`, then `unit` and `internalItems` are left as-is, which seems like an issue.

Reply:
I've cleared internalItems. Assigning a nil schedule after the initial setting is not expected. I think keeping units changed is ok.